File: src/1_100/0055_jump-game/jump-game.java
public class Solution {
    public boolean canJump(int[] nums) {
		if(nums==null) {
			return true;
		}
		int n = nums.length;
		int[] memo = new int[n];
		memo[n-1]=1;
		for(int i=n-2;i>=0;--i) {
			int can_jump = Math.min(nums[i]+i,n-1);
			for(int j=i+1;j<=can_jump;++j) {
				if(memo[j]==1) {
					memo[i] = 1;
					break;
				}
			}
		}
		return memo[0]==1;
    }
}

File: src/1_100/0055_jump-game/jump-game.py
class Solution(object):
	def canJump(self, nums):
		if not nums:
			return True
		max_jump = 0
		n = len(nums)
		for i in xrange(n):
			if max_jump>=i:
				max_jump = max(max_jump,i+nums[i])
		if max_jump>=n-1:
			return True
		return False